### PR TITLE
[FIX](map/struct)fix map/struct literal from fe

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java [hunk 1]
```diff
@@ -240,6 +251,8 @@ public abstract class Type {
         structSubTypes.add(DECIMAL32); // same DEFAULT_DECIMALV3
         structSubTypes.add(DECIMAL64);
         structSubTypes.add(DECIMAL128);
```
Comment (morrySnow) [inline fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:253]: decimal256?

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java [hunk 2]
```diff
@@ -249,6 +262,10 @@ public abstract class Type {
         structSubTypes.add(CHAR);
         structSubTypes.add(VARCHAR);
         structSubTypes.add(STRING);
+        structSubTypes.add(NULL);
+        structSubTypes.add(ARRAY);
+        structSubTypes.add(MAP);
+        structSubTypes.add(STRUCT);
```
Comment (morrySnow) [inline fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:268]: do we support variant?

**Comment by dataroaring** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:253]: temp not support

File: fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java (hunk 2 above)
Comment (amorynan) [reply fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java:268]: no

**Comment by amorynan** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by mymeiyi — APPROVED** [review]

**Pull request merged**